docs/smtchecker.rst

```diff
@@ -404,16 +404,60 @@ that the assertion fails:
      | 		^^^^^^^^^^^^^^^^^
 
 
+******************
+SMTChecker Options
+******************
+
+Model Checking Engines
+======================
+
+The SMTChecker module implements two different reasoning engines, a Bounded
+Model Checker (BMC) and a system of Constrained Horn Clauses (CHC).  Both
+engines are currently under development, and have different characteristics.
+The engines are independent and every property warning states from which engine
+it came from. Notice that all the examples above with counterexamples were
+reported by CHC, the more powerful engine.
+
+By default both engines are used, where CHC runs first, and every property that
+was not proven is passed over to BMC. You can choose a specific engine via the CLI
+option ``--model-checker-engine {all,bmc,chc,none}`` or the JSON option
+``settings.modelChecker.engine={all,bmc,chc,none}``.
+
+Bounded Model Checker (BMC)
+---------------------------
+
+The BMC engine analyzes functions in isolation, that is, it does not take the
+overall behavior of the contract throughout many transactions into account when
+analyzing each function.  Loops are also ignored in this engine at the moment.
+Internal function calls are inlined as long as they are not recursive, direct
```

Comment:
direct*ly*
I'd also make it "as long as they are not (directly or indirectly) recursive."

Reply:
I think you can omit the suffix here if the next word has it, here "direct or indirectly"

Reply:
Never heard that one before :-D...

Reply:
The grammar processor in my brain highly disagrees in any case :-).

Reply:
`and :: 'a => 'a => 'a`, `direct :: adjective`, `indirectly :: adverb`,
`(and) direct indirectly` - clash of types :-).

Reply:
To me it would seem the example either reads "(direct) and (indirectly liquid-cooled) coreless linear actuators" or (maybe in that particular case more likely) is also just wrong :-).

Reply:
But yeah for the record: fine with keeping it, even though I think it's wrong :-).

Reply:
I haven't found proof, and your case seems more consistent so I'll change it :)